internal/dangling/dangling_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dangling
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
+	nodegroupv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1/helper/nodegroup"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	nrouid  = "123"
+	ns      = "nro-ns"
+	nroName = "nro"
+)
+
+var (
+	targetMCPName = "mcp1" // var because we need its pointer
+	ctx           = context.TODO()
+
+	nro = nropv1.NUMAResourcesOperator{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nroName,
+			Namespace: ns,
+			UID:       nrouid,
+		},
+	}
+
+	trees = []nodegroupv1.Tree{
+		{
+			NodeGroup: &nropv1.NodeGroup{
+				PoolName: &targetMCPName,
+			},
+			MachineConfigPools: []*mcov1.MachineConfigPool{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: targetMCPName,
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestDeleteUnusedMachineConfigs(t *testing.T) {
+	mcs := []mcov1.MachineConfig{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("51-%s-%s", nroName, targetMCPName),
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						UID: nrouid,
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("51-%s-mcp2", nroName),
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						UID: nrouid, // owned by NRO object but dangling
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "51-XX-mcp1",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						UID: "100", // not owned by NRO object
+					},
+				},
+			},
+		},
+	}
+
+	_ = mcov1.AddToScheme(scheme.Scheme)
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(&mcs[0], &mcs[1], &mcs[2]).Build()
```

Comment:
if we turn `mcs` in a slice of pointers, we can just write `WithObjects(mcs...)`

Reply:
that doesn't work actually with implemented interface of type client.Object